src/front/mod.rs

```diff
@@ -277,6 +277,15 @@ where
     pub fn add(&mut self, name: Name, var: Var) -> Option<Var> {
         self.scopes[self.cursor - 1].insert(name, var)
     }
+
+    /// Adds a new variable to the root scope.
+    ///
+    /// Returns the previous variable with the same name in the root scope if it
+    /// exists, so that the frontend might handle it in case variable shadowing
+    /// is disallowed.
```

Comment:
It might be nice to briefly say something like:

> This is used for certain kinds of declarations in GLSL, like yadda yadda, which introduce definitions into the global scope, not the innermost scope enclosing their declaration.

or something like that.